macro-mathjax-ui/src/main/resources/Macros/MathJaxMacro.xml

```diff
@@ -187,6 +174,49 @@ y = \frac{c}{cb-ad}
       load: ['[tex]/noerrors']
     }
   };
+
+  /**
+   * Reset the tex labels, since re-typesetting might create duplicate labels, and perform an asynchronous typesetting.
+   * See https://docs.mathjax.org/en/latest/web/typeset.html#handling-asynchronous-typesetting.
+   */
+  var clearAndTypesetPromise = function(elems) {
+    let typesetPromise = Promise.resolve();
+    MathJax.texReset();
+    if (elems != undefined) {
+      // This could lead to bugs https://github.com/mathjax/MathJax/issues/2999, so for now typesetting specific
+      // elements should be used only if equations numbering does not count. For example, this is used for typesetting
+      // in rendered diff view, since only one math element is send at a time.
+      typesetPromise = MathJax.typesetPromise(elems);
+    } else {
+      typesetPromise = MathJax.typesetPromise();
+    }
+    typesetPromise.catch((err) =&gt; console.log('Typeset failed: ' + err.message));
+    return typesetPromise;
+  };
+
+  /**
+   * Since the same macro will be displayed twice in a rendered diff view, we typeset each one individually in order to
+   * avoid errors from duplicate labels. This will produce some inconsistences with the displayed equation numbers,
+   * meaning that each macro will start counting from 1 and anchor ids will be duplicated.
+   */
+  var typesetInRenderedDiffViewPromise = function() {
+    let promise = Promise.resolve();
+    $('#renderedChanges .xwiki-mathjax').each((i, el) =&gt; {
+      if ($(el).find('mjx-container').length &gt; 0) {
+        return;
+      }
+      // Mathjax doesn't typesets math code mixed with html content.
+      el.innerHTML = el.textContent;
+      promise = promise.then(() =&gt; clearAndTypesetPromise([el]));
```

Comment:
So ``clearAndTypesetPromise`` is called twice, once for the specific element and a second time for all elements. I assume this is the desired behaviour.

Reply:
Yes. The second typeset is needed for math elements which weren't already typeset (outside the `renderedDiff` block) . This will make the rest of the math elements in the page 'act normal', as if doing a clean typeset of the page